pkg/kube/interface.go

```diff
@@ -34,6 +34,8 @@ type Interface interface {
 
 	WaitWithJobs(resources ResourceList, timeout time.Duration) error
 
+	WaitForDelete(resources ResourceList, timeout time.Duration) error
```

Comment:
IIRC from conversations with @technosophos on this... Changes to an `interface` breaks compatibility according to Go conventions. So this should be a separate interface if possible.

Think of it this way: if I implement a type `Reader` with func `Read() []byte`, but then a future update introduces `ReadString() string` to the `Reader` interface, the user now is forced to implement `ReadString() string` in their own types to satisfy the interface.

Reply:
Thank you for the review and feedback. 
I am ok with adding a new separate interface but could you please see the PR https://github.com/helm/helm/pull/8363 where the interface was updated with a new `WaitWithJobs` function?  Since the `WaitForDelete` in this PR is similar to the `WaitWithJobs`, can an exception be made for this PR as well?